### PR TITLE
QFunction - add QFunctionView

### PR DESCRIPTION
**Pull request by jeremylt**
I can also add an OperatorView if desirable, its the only one left out.

### CODE REVIEW

**Comment by jeremylt** [comment]
Huh, TIL C++ hates designated initializers.

@jedbrown, I'm trying to figure out how PETSc provides these arrays for printing, such as `PCMGCycleTypes` and `PCMGTypes`, but I don't see where the arrays are actually populated. Any insight?
I just see empty? declarations here.
https://www.mcs.anl.gov/petsc/petsc-dev/include/petscpc.h.html

**Comment by jedbrown** [comment]
In source files, e.g.,
https://www.mcs.anl.gov/petsc/petsc-dev/src/ksp/pc/impls/mg/mg.c.html#line490

File: include/ceed.h [hunk 1]
```diff
@@ -148,6 +148,11 @@ typedef enum {
   CEED_MEM_DEVICE,
 } CeedMemType;
 
+static const char *const CeedMemTypes[] = {
```
Comment (jedbrown) [inline include/ceed.h]: Yeah, make this `extern` and just a declaration, then define it in a file.

Comment (jeremylt) [reply include/ceed.h]: Oh. Duh.

Thank you

File: tests/output/t523-operator-f.out [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    [0] Operator Field "_weight"
+      Lmode "no transpose"
+      No vector
+    [1] Operator Field "dx"
+      Lmode "no transpose"
+      Active vector
+  1 Output Field
+    [0] Operator Field "_weight"
+      Lmode "no transpose"
+      Collocated basis
```
Comment (jeremylt) [inline tests/output/t523-operator-f.out]: I welcome any feedback into the exact appearance of these outputs, but now we can view the basic data about any Ceed object.

Comment (valeriabarra) [reply tests/output/t523-operator-f.out]: I like it!
I would personally only put colons to explain the following list of objects, like
```
SubOperator [0]:
  3 Fields
  2 Input Fields:
```
Good job!

Comment (jeremylt) [reply tests/output/t523-operator-f.out]: Cool, how about now?

**Comment by jeremylt** [comment]
Sample User QFunction:
```
User CeedQFunction 
  2 Input Fields:
    Input Field [0]:
      Name: "qdata"
      Size: 1
      EvalMode: "interpolation"
    Input Field [1]:
      Name: "u"
      Size: 1
      EvalMode: "interpolation"
  1 Output Field:
    Output Field [0]:
      Name: "v"
      Size: 1
      EvalMode: "interpolation"
```
Sample Gallery QFunction:
```
Gallery CeedQFunction MassApply
  2 Input Fields:
    Input Field [0]:
      Name: "u"
      Size: 1
      EvalMode: "interpolation"
    Input Field [1]:
      Name: "qdata"
      Size: 1
      EvalMode: "none"
  1 Output Field:
    Output Field [0]:
      Name: "v"
      Size: 1
      EvalMode: "interpolation"
```
Sample Operator:
```
CeedOperator
  3 Fields
  2 Input Fields:
    Input Field [0]:
      Name: "_weight"
      Lmode: "no transpose"
      No vector
    Input Field [1]:
      Name: "dx"
      Lmode: "no transpose"
      Active vector
  1 Output Field:
    Output Field [0]:
      Name: "_weight"
      Lmode: "no transpose"
      Collocated basis
      Active vector
```
Sample Composite Operator:
```
Composite CeedOperator
  SubOperator [0]:
    3 Fields
    2 Input Fields:
      Input Field [0]:
        Name: "_weight"
        Lmode: "no transpose"
        No vector
      Input Field [1]:
        Name: "dx"
        Lmode: "no transpose"
        Active vector
    1 Output Field:
      Output Field [0]:
        Name: "_weight"
        Lmode: "no transpose"
        Collocated basis
  SubOperator [1]:
    3 Fields
    2 Input Fields:
      Input Field [0]:
        Name: "_weight"
        Lmode: "no transpose"
        No vector
      Input Field [1]:
        Name: "dx"
        Lmode: "no transpose"
        Active vector
    1 Output Field:
      Output Field [0]:
        Name: "_weight"
        Lmode: "no transpose"
        Collocated basis
```

File: tests/output/t523-operator-f.out (hunk 2 above)
Comment (valeriabarra) [reply tests/output/t523-operator-f.out]: It helps with my OCD 😁 but it was really ok if you preferred the other way! Thanks. Nice job again.

**Comment by jeremylt** [comment]
Will merge after Travis + CodeCov.

**Pull request merged**